pkg/db/query_gorm.go

```diff
@@ -0,0 +1,99 @@
+package db
+
+import (
+	"go.uber.org/zap"
+	gormPg "gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type QueryGorm struct {
+	Conn          Connection
+	gormDB        *gorm.DB
+	columnHandler *ColumnHandler
+	DbEntity      DatabaseEntity
+}
+
+func NewQueryGorm(conn Connection, entity DatabaseEntity, logger *zap.SugaredLogger) (*QueryGorm, error) {
```

Comment:
Currently you create new GORM sessions per query multiple times even. I think this is very bad for Performance if we use the default settings. I think we should either create one shared session from the Connection or create one on-demand through an interface method. Right now the danger of misuse is too high imo.

Reply:
I totally get your point! Tbh in the way I configured it in this PR at the moment with using the `DryRun` mode, we should not get any performance issues, since we are still using our `Connection` interface for the queries. Thus, the connection of the gorm session will not be used atm.

More in my other comment: https://github.com/kyma-incubator/reconciler/pull/1029#discussion_r849261031

But I can also rewrite it as a preparation for the follow-up.

Reply:
I agree with your conclusion and think we should not include this change right now but lets follow-up with a lower-prio ticket. This for sure still fixes the issues with the DB Query Building!